Dockerfile

```diff
@@ -1,30 +1,23 @@
-FROM debian:jessie
-
-RUN apt-get update && apt-get install -y \
-    build-essential \
-    curl \
-    nano \
-    supervisor
-
-# Install Node.js
-RUN curl -sL https://deb.nodesource.com/setup_4.x | bash -
-RUN apt-get install -y nodejs
-
-# Improve cache invalidations by only running npm if requirements have indeed changed
-WORKDIR /app
-COPY package.json /app/
-RUN npm install
-
-# Supervisor settings
-COPY docker/supervisord.conf /etc/supervisor/conf.d/atlas.conf
+FROM node:4-onbuild
 
 # Application source code
 COPY config/ /app/config
-COPY src/ /app/src
 COPY index.js /app/
 COPY scripts.js /app/
+COPY package.json /app/
+
+COPY src/ /app/src
+WORKDIR /app/src
+
+# Install node-gyp as global and ensure it's all clean and tide
+RUN npm install -g node-gyp && \
+    node-gyp clean && \
+    npm cache clean
+
+RUN npm install
 
+ENV NODE_ENV development
 # Expose application server port
 EXPOSE 8000
 
-CMD ["supervisord", "-c", "/etc/supervisor/supervisord.conf", "-n"]
+CMD ["npm", "run", "dev"]
```

Comment:
This Dockerfile is intended to be used in any environment, not only `development`